src/linalg.jl

```diff
@@ -149,3 +149,14 @@ function _mul!(C::BlockDiagonal, A::BlockDiagonal, B::BlockDiagonal, α::Number,
 
     return C
 end
+
+function LinearAlgebra.:\(A::BlockDiagonal, B::AbstractVecOrMat)
+    i = 1
+    c = similar(B)
+    for a in blocks(A)
+        d = size(a, 1)
+        c[i:i+d-1, :] = a \ B[i:i+d-1, :]
+        i += d
+    end
+    return c
+end
```

Comment:
```suggestion
function LinearAlgebra.:\(B::BlockDiagonal, vm::AbstractVecOrMat)
    row_i = 1
    result = similar(vm)
    for block in blocks(B)
        nrow = size(block, 1)
        result[row_i:row_i+nrow-1, :] = block \ vm[row_i:row_i+nrow-1, :]
        row_i += nrow
    end
    return result
end
```
May I suggest calling the variables differently for easier reading?

The only other question is does this work when the blocks are not squared matrices?

Reply:
Sure! I am adding your changes.

No it would not work, I did not even realize you could create a `BlockDiagonal` out of non-square matrices, I will add an initial check at the beginning